### ISSUE TITLE
TTNN pow op calculates incorrect results for input > 1

### ISSUE THREAD

**Issue by alyashev** [tenstorrent/tt-metal#19821]
**Describe the bug**
ttnn.pow(9,2) != 81

**To Reproduce**
import torch
import ttnn
import numpy as np

torch.manual_seed(0)

device_id = 0
device = ttnn.open_device(device_id=device_id)
ttnn.enable_program_cache(device)

torch_a = torch.randn((1), dtype=torch.float32) #bfloat16)
torch_b = torch.randn((1), dtype=torch.float32) #bfloat16)
torch_a[0] = (-9)
torch_b[0] = (2)

a = ttnn.from_torch(torch_a, layout=ttnn.TILE_LAYOUT, device=device)
b = ttnn.from_torch(torch_b, layout=ttnn.TILE_LAYOUT, device=device)
a_torch = a.cpu().to_torch()
b_torch = b.cpu().to_torch()
#np_a  = a_torch.cpu().numpy()
#np_b  = b_torch.cpu().numpy()

output = ttnn.pow(a, b);
torch_output = torch.pow(torch_a,torch_b)

output = ttnn.to_layout(output, ttnn.ROW_MAJOR_LAYOUT)

print("Printing ttnn tensor")
print(f"shape: {output.shape}")

output_torch = output.cpu().to_torch()
#out_np = np.power(np_a, np_b)

#print(a_torch[0], "^", b_torch[0], " torch: ", torch_output[0], " ttnn:  ", output_torch[0], " numpy: ", out_np[0])
print(a_torch[0], "^", b_torch[0], " torch: ", torch_output[0], " ttnn:  ", output_torch[0])

ttnn.close_device(device)

**Expected behavior**
9^2 == 81

**Screenshots**

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Version of software (eg. commit)

**Additional context**

**Comment by umadevimcw** [comment]
@alyashev Can you whether it is `-9` or `9`?

In the description you mentioned pow(9,2) but in the example it given as -9

**Comment by alyashev** [comment]
have you tried the script?
If so, could you please change the sign?
any value with abs > 1 is wrong.
Regards.

**Comment by alyashev** [comment]
The verification framework is not correct. it always returns PASS. good for you.

**Comment by cmaryanTT** [comment]
@nvelickovicTT - the test on our side was problematic; and for whatever reason it was never caught in a model.

**Comment by nvelickovicTT** [comment]
@rdjogoTT this is part of the SFPU binary support you added end of last year. Which tests were you running to verify? How was accuracy evaluated?

**Comment by umadevimcw** [comment]
@rdjogoTT @cmaryanTT Algorithm mentioned here https://gist.github.com/jrade/293a73f89dfef51da6522428c857802d seems working fine. Even in TT implementation its mentioned same algoirthm. We can cross verify with this.

**Comment by fvranicTT** [comment]
I compiled a list of results based on the above program:
[pow_precision.xlsx](https://github.com/user-attachments/files/19737218/pow_precision.xlsx)
It shows absolute/relative errors of x^y operations on a range of inputs.

**Comment by cmaryanTT** [comment]
@nmauriceTT reviewed the accuracy of exp/pow (and many other functions) and will be looking at the implementation to see what the causes of our problems are and what the fix should be.

**Comment by nvelickovicTT** [comment]
@nmauriceTT kind ping to share the document with accuracy reviewed.

**Comment by nvelickovicTT** [comment]
Removed LLK label as our team is not working actively on this.

**Comment by cmaryanTT** [comment]
We have a prototype new version of pow() that gets within 1-2 ULP for most reasonable values. Testing now.

**Comment by cmaryanTT** [comment]
More test cases in https://github.com/tenstorrent/tt-metal/issues/23529 - closing that case as a duplicate of this one.

**Comment by fahmedchanTT** [comment]
@nmauriceTT Are we able to close this case? The referenced MRs have been merged.

**Comment by nmauriceTT** [comment]
@fahmedchanTT The problem has been solved for Wormhole but not yet for Blackhole.
We can close the case once [this PR](https://github.com/tenstorrent/tt-metal/pull/26270) gets merged for Blackhole (a compiler bug prevented the port, but it was fixed last night)
